src/parser.yy

```diff
@@ -286,12 +289,19 @@ expr : int             { $$ = $1; }
      | "(" IDENT MUL ")" expr %prec CAST  { $$ = new ast::Cast($2, true, $5, @1 + @4); }
      ;
 
-ident : IDENT   { $$ = $1; }
-      | BUILTIN { $$ = $1; }
+ident : IDENT         { $$ = $1; }
+      | BUILTIN       { $$ = $1; }
+      | CALL          { $$ = $1; }
+      | CALL_BUILTIN  { $$ = $1; }
+      | STACK_MODE    { $$ = $1; }
       ;
 
-call : ident "(" ")"       { $$ = new ast::Call($1, @$); }
-     | ident "(" vargs ")" { $$ = new ast::Call($1, $3, @$); }
+call : CALL "(" ")"               { $$ = new ast::Call($1, @$); }
+     | CALL "(" vargs ")"         { $$ = new ast::Call($1, $3, @$); }
+     | CALL_BUILTIN  "(" ")"      { $$ = new ast::Call($1, @$); }
+     | CALL_BUILTIN "(" vargs ")" { $$ = new ast::Call($1, $3, @$); }
+     | ident "(" ")"              { error(@1, "Unknown function: " + $1); YYERROR;  }
+     | ident "(" vargs ")"        { error(@1, "Unknown function: " + $1); YYERROR;  }
```

Comment:
These two lines are introducing more shift/reduce conflicts into the parser, since `CALL()` can either take the rule on line 299 or the rule on 303, for example

Reply:
good point. I've updated it. Its a bit verbose but I didn't want to modify too much in 1 go.